tests/test_peaklet_processing.py

```diff
@@ -79,3 +83,45 @@ def test_tight_coincidence(hits, channel):
         m_hits_in_peak &= (hits_max_time <= (p_max_t + right))
         n_channel = len(np.unique(hits[m_hits_in_peak]['channel']))
         assert n_channel == tight_coin_channel[ind], f'Wrong number of tight channel got {tight_coin_channel[ind]}, but expectd {n_channel}'  # noqa
+
+
+class TestDataCLass(unittest.TestCase):
+    """
+    Base class which downloads test data and makes data up-to the
+    highest data_type specfied. The data can be used in tests.
+    """
+    highest_data_type = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Common setup for all the tests. We need some data which we
+        don't delete but reuse to prevent a lot of computations in this
+        class.
+        """
+        cls.st = nt_test_context()
+        if cls.highest_data_type:
+            cls.st.make(nt_test_run_id, cls.highest_data_type)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        """
+        Removes test data after tests are done.
+        """
+        path = os.path.abspath(cls.st.storage[-1].path)
+        for file in os.listdir(path):
+            shutil.rmtree(os.path.join(path, file))
+
+
+class TestPeaklets(TestDataCLass):
+    """
+    Test for different peaklet properties.
+    """
+    highest_data_type = 'peaks'
+
+    def test_area_fraction_top(self):
+        assert self.st.is_stored(nt_test_run_id, 'merged_s2s')
+
+        merged_s2s = self.st.get_array(nt_test_run_id, 'merged_s2s')
+        _area_close_to_area_per_channel = np.isclose(merged_s2s['area']/np.sum(merged_s2s['area_per_channel'], axis=1), 1)
+        assert np.all(_area_close_to_area_per_channel)
```

Comment:
@JoranAngevaare I added now this base TestDataClass which we can inherit and use in different tests. This should unify things a bit. Since it depends on unittest I was not sure where to put it. Best place might be test_utils, any different opinion?

Reply:
Right, but this is going to be way to slow if we remove the data each time to be useful since we build it from raw-records

Reply:
But how do we intend to catch any bugs in processing if we are just loading already stored data all the time? Those would be only caught on github.

Reply:
I don't oppose this, however, it simply would destroy the testing times.

 We could very easily make the distinction on github actions by setting an environment variable on github
`STORE_TEST_OUTCOMES` that would disable the `tearDownClass` method you wrote here.

I did similar things before:
https://github.com/XENONnT/straxen/blob/master/.github/workflows/pytest.yml#L104-L105